Added default download path

NejcZdovc opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Resolves #2110

## Auditors
@bradleyrichter @bsclifton

## Test Plan
- go to the preferences, general tab
- define your default download path
- toggle show dialog
- based on your settings, download dialog should be opened or not
- if you set to hide download dialog, files should be downloaded automatically to your default download path

NejcZdovc commented:
@bradleyrichter in #2110 you added pencil at the end of the input. Can you please provide an icon for it, if we will use it. For now I set input to read only and when you click on it, dialog is shown automatically.

NejcZdovc commented:
@bsclifton Sorry I added you as a review too soon. My mistake. This PR is not ready for a final review as you notice there is no test for this new feature. I added label `ready-for-review` so that @bradleyrichter can look through it and prepare icon for browse/edit. And yes I agree we need to have a browse/edit icon.

bradleyrichter commented:
@NejcZdovc SVG icon for the edit (pencil) button attached:


[browser_icon_pencil.svg.zip](https://github.com/brave/browser-laptop/files/765198/browser_icon_pencil.svg.zip)

bsclifton commented:
@NejcZdovc was this one done?

NejcZdovc commented:
@bsclifton I added test's as well, so we are ready for a review

bsclifton commented on app/browser/reducers/downloadsReducer.js:
```diff
@@ -89,6 +92,18 @@ const downloadsReducer = (state, action) => {
+        properties: ['openDirectory']
+      }, (folder) => {
+        if (Array.isArray(folder) && fs.lstatSync(folder[0]).isDirectory()) {
+          appActions.changeSetting(settings.DOWNLOAD_DEFAULT_PATH, folder[0])
```
I think this is fine as-is... but it could be better as it's own discreet action (in the event that we'd like other reducers to be listening).  Perhaps an action like `defaultDownloadPathChanged`

NejcZdovc commented:
@bsclifton First of all sorry, I somehow over looked your second review.

I added hover color effect. Regarding the pointer I remember talking with @bradleyrichter about it for bookmark manager and he said that we should use only color hover and not a pointer. Bookmark manager for example don't have single pointer hover, only color hover. I agree that we should be consistent, so we need a guide line for it. 

For user select I suggest to add it after #7481 is merged, because I added aphrodite param for component `SettingItem`.

NejcZdovc commented:
Just for a reference, @bridiver added flag to handle "Save as" calls https://github.com/brave/muon/commit/de90c69d2a1432bedf8f43f4fa5498bcb56a4655

bridiver commented on js/about/aboutActions.js:
```diff
@@ -407,6 +407,15 @@ const aboutActions = {
+  /**
+   * Open dialog for default download path setting
+   */
+  defaultDownloadPath: function () {
```
these no longer have to go in `aboutActions` and this file should be deprecated. Use `appActions` for now instead

NejcZdovc replied:
Moved

bridiver commented:
this will need changes to correspond with my muon updates to correctly handle the prompt

NejcZdovc commented:
@bsclifton added save prompt for save as. Ready for a review

bsclifton commented:
An area we can improve (cc: @bradleyrichter): 

If I have a long path, it gets clipped
![screen shot 2017-03-16 at 2 56 24 pm](https://cloud.githubusercontent.com/assets/4733304/24020396/3a2bbef8-0a59-11e7-9f12-20285b550e8e.png)

It looks the exact same in Chrome though, so not a deal-breaker or anything... just an area we could do better 😄 
![screen shot 2017-03-16 at 2 57 51 pm](https://cloud.githubusercontent.com/assets/4733304/24020419/4fec8ec0-0a59-11e7-99e4-6a368e9bb003.png)
